src/libutil/file-ingestion-method.cc

```diff
@@ -0,0 +1,18 @@
+#include "file-ingestion-method.hh"
+#include "error.hh"
+
+namespace nix {
+
+std::string makeFileIngestionPrefix(FileIngestionMethod m)
```

Comment:
I don't really want this sort of encoding thing to move from `libstore`. it is an arbitrary choice of the store layer, not a reusable mechanism. If anything, I want to move this stuff out of `content-address.{cc,hh}` and back to the places where it is used (printing and parsing derivations, store paths) since it isn't really consistent between those two things.

Reply:
It seems that `InputAccessor` would have to expose a fingerprint, so that the rest of `fetchToStore` can be moved back to libstore.

Reply:
Agreed, let me see how much of this I can move back to `libstore`.

Reply:
OK, I moved `file-content-address.hh` and `file-ingestion-method.hh` from `libutil` to `libstore`.

Reply:
Sorry I was unclear. I *do* want the enums themselves to live in `libutil`. Those do belong with `archive.hh` and `source-accessor.hh`, which are also in `libutil`. What doesn't belong is *just* the prefix string bits. The string prefix bits are *not* intrinsic to these concepts, and just relate to downstream concerns like derivation aterm and narinfo CA field.

If like, perhaps take a stab at https://github.com/NixOS/nix/issues/9278 ? The point of that issue to was to make precisely this sort of thing clear:

 - "File System Object" is a barely nix-specific notation, and that can stay in `libutil` (originally I proposed it go in a new library below `libstore` also.

 - "Store Object" is a very much nix-specific concept that should stay in `libstore`.

The prefix stuff in question have have a file system object name, but it is neither intrinsic to "file system object" or "store object". I should just push it down to the derivation / store path / narinfo serializations that need it. Serializations are intrinsic properties of data types being serialized, an indeed this colon-prefix stuff is no good and not not be used in the newer unstable JSON serializations. (It currently is, this needs to change.)

Reply:
The short version I that you can leave things the way they were prior to this PR if you like, and I'll at some future point I'll move this serialization nonsense out of `content-address.cc`.